pallets/basic-sto/src/lib.rs

```diff
@@ -0,0 +1,174 @@
+// Copyright (c) 2020 Polymath
+
+//! # STO Module
+
+#![cfg_attr(not(feature = "std"), no_std)]
+#![recursion_limit = "256"]
+
+use codec::{Decode, Encode};
+use frame_support::{
+    decl_error, decl_event, decl_module, decl_storage, dispatch::DispatchResult, ensure,
+};
+use frame_system::{self as system, ensure_signed};
+use pallet_identity as identity;
+use pallet_settlement::{Leg, SettlementType};
+use polymesh_common_utilities::{
+    traits::{asset::Trait as AssetTrait, identity::Trait as IdentityTrait, CommonTrait},
+    Context,
+};
+use polymesh_primitives::{IdentityId, Ticker};
+use sp_runtime::traits::{CheckedMul, Saturating};
+use sp_std::prelude::*;
+
+type Identity<T> = identity::Module<T>;
+type Settlement<T> = pallet_settlement::Module<T>;
+
+pub trait Trait:
+    frame_system::Trait + CommonTrait + IdentityTrait + pallet_settlement::Trait
+{
+    /// The overarching event type.
+    type Event: From<Event<Self>> + Into<<Self as frame_system::Trait>::Event>;
+}
+
+/// Details about the Fundraiser
+#[cfg_attr(feature = "std", derive(Debug))]
+#[derive(Encode, Decode, Default, Clone, PartialEq, Eq, PartialOrd, Ord)]
+pub struct Fundraiser<Balance> {
+    /// Token to raise funds in
+    pub raise_token: Ticker,
+    /// Amount of offering token available for sale
+    pub remaining_amount: Balance,
+    /// Price of one million offering token units (one full token) in terms of raise token units
+    pub price_per_token: u128,
+    /// Id of the venue to use for this fundraise
+    pub venue_id: u64,
```

Comment:
Consider newtyping this or a type alias for clarity throughout.

Reply:
Perhaps in a different PR.